Alloy/Encoders/MaxEncoder.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  MaxEncoder.swift
+//  Adjuster
+//
+//  Created by Eugene Bokhan on 13/02/2019.
+//
+
+import Metal
+import simd
+
+public class MaxEncoder {
+
+    enum Errors: Error {
+        case functionCreationFailed
+        case encoderCreationFailed
+        case invalidBufferStorageMode
+    }
+
+    let pipelineState: MTLComputePipelineState
+
+    public init(library: MTLLibrary) throws {
+        self.pipelineState = try library.computePipelineState(function: MaxEncoder.functionName)
+    }
+
+    public func encode(inputTexture: MTLTexture,
+                       blockSize: BlockSize,
+                       resultBuffer: MTLBuffer,
+                       in commandBuffer: MTLCommandBuffer) throws {
+        commandBuffer.compute { (encoder) in
+            encoder.setTexture(inputTexture, index: 0)
+            encoder.set(blockSize, at: 0)
+            encoder.setBuffer(resultBuffer,
+                              offset: 0,
+                              index: 1)
+            let threadgroupSize = self.pipelineState.max2dThreadgroupSize
```

Comment:
I really think that's too much. needs testing of course, but I think it will be slower, than having something like 8x8 grid